datafusion/expr/src/type_coercion/binary.rs

```diff
@@ -514,7 +514,7 @@ pub fn coercion_decimal_mathematics_type(
                 left_decimal_type,
                 right_decimal_type,
             ),
-            Operator::Multiply | Operator::Divide | Operator::Modulo => {
+            Operator::Divide | Operator::Modulo => {
                 get_wider_decimal_type(left_decimal_type, right_decimal_type)
             }
```

Comment:
Huh? This is not correct...

Reply:
Why decimal multiply decimal need to cast to a common wider type ? I do not change the get result type logic.
I think in the latest SparkSQL,  for decimal multiply, they do not cast to common wider type either.

Reply:
My understanding is it just need to multiply the internal i128 integers, and cast the result back to the required result 
precision and scale.

Reply:
Hm, I remember this follows Spark decimal coercion rule to cast lhs/rhs to wider decimal type. I don't see Spark does this now for multiply.

Yea, you should be correct. We don't need to cast lhs/rhs to wider decimal type for multiply.

It's too late. Let me think it again tomorrow morning. 😄

Reply:
Rethink about it. The issue is not to follow Spark or not in this. But for multiply, we don't need to align two integers.